### ISSUE TITLE
logger: Allow logging to both STDERR and STDOUT

### ISSUE THREAD

**Issue by nareddyt** [envoyproxy/envoy#8620]
The [command line documentation](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-log-path) contains this behavior for `--log-path`. I have confirmed this documentation is accurate via manual testing.

> (optional) The output file path where logs should be written. This file will be re-opened when SIGUSR1 is handled. If this is not set, log to stderr.

Can we support writing logs to both STDOUT and STDERR when this flag is not set? It would be great if `error` and `critical` logs can be directed to STDERR, and all other lower-level logs be directed to STDOUT. I believe this is a standard practice.

I'm happy to implement the change, but some direction would be helpful.

**Comment by nareddyt** [comment]
BTW this would be helpful for stackdriver logging integration. Since all logs output to STDERR, stackdriver defaults them to error level. This makes it hard to filter through and find errors in the console.

**Comment by nareddyt** [comment]
FYI this issue is about application logs, not access logs.

It seems like the current implementation is to write access logs to STDOUT, and application logs to STDERR.

I did some more reading on best practices, and perhaps the current implementation is expected: 
- Conventional output = access logs = STDOUT
- Diagnostic output = application logs = STDERR

https://unix.stackexchange.com/a/331620/331278

Closing this issue, will look into structured logging instead.
